userapi/storage/accounts/sqlite3/storage.go

```diff
@@ -124,8 +143,12 @@ func (d *Database) CreateGuestAccount(ctx context.Context) (acc *api.Account, er
 	// when the first txn upgrades to a write txn. We also need to lock the account creation else we can
 	// race with CreateAccount
 	// We know we'll be the only process since this is sqlite ;) so a lock here will be all that is needed.
-	d.createAccountMu.Lock()
-	defer d.createAccountMu.Unlock()
+	d.profilesMu.Lock()
+	d.accountDatasMu.Lock()
+	d.accountsMu.Lock()
+	defer d.profilesMu.Unlock()
+	defer d.accountDatasMu.Unlock()
+	defer d.accountsMu.Unlock()
```

Comment:
I don't like how we have so many mutexes. We run the risk of deadlocking if we lock in the wrong order for `CreateGuestAccount` and  `CreateAccount`, we should at least comment that here.

Reply:
The alternative is that we just have a single lock for all write operations?

Reply:
That would work, but this works too.